compiler/tests/evm_contracts.rs

```diff
@@ -0,0 +1,321 @@
+
+        harness.test_function(
+            &mut executor,
+            "write_bar",
+            vec![u256_token(4), u256_token(42)],
+            None,
+        );
+
+        harness.test_function(
+            &mut executor,
+            "write_bar",
+            vec![u256_token(420), u256_token(12)],
+            None,
+        );
+
+        harness.test_function(
+            &mut executor,
+            "read_bar",
+            vec![u256_token(4)],
+            Some(u256_token(42)),
+        );
+
+        harness.test_function(
+            &mut executor,
+            "read_bar",
+            vec![u256_token(420)],
+            Some(u256_token(12)),
+        );
+    })
+}
+
+#[test]
+fn address_bytes10_map() {
+    with_executor(&|mut executor| {
+        let harness = deploy_contract(&mut executor, "address_bytes10_map.vy", "Foo");
+
+        let address1 = address_token("0000000000000000000000000000000000000001");
+        let bytes1 = bytes_token("ten bytes1");
+
+        let address2 = address_token("0000000000000000000000000000000000000002");
+        let bytes2 = bytes_token("ten bytes2");
+
+        harness.test_function(
+            &mut executor,
+            "write_bar",
+            vec![address1.clone(), bytes1.clone()],
+            None,
+        );
+
+        harness.test_function(
+            &mut executor,
+            "write_bar",
+            vec![address2.clone(), bytes2.clone()],
+            None,
+        );
+
+        harness.test_function(&mut executor, "read_bar", vec![address1], Some(bytes1));
+
+        harness.test_function(&mut executor, "read_bar", vec![address2], Some(bytes2));
+    })
+}
+
+#[test]
+fn guest_book() {
+    with_executor(&|mut executor| {
+        let mut harness = deploy_contract(&mut executor, "guest_book.vy", "GuestBook");
+
+        let sender = address_token("1234000000000000000000000000000000005678");
+        let bytes = bytes_token(
+            iter::repeat("ten bytes.")
+                .take(10)
+                .collect::<String>()
+                .as_str(),
+        );
+
+        harness.caller = sender.clone().to_address().unwrap();
+
+        harness.test_function(&mut executor, "sign", vec![bytes.clone()], None);
+
+        harness.test_function(&mut executor, "get_msg", vec![sender], Some(bytes.clone()));
+
+        // Destruct the executor
+        harness.event_emitted(executor, "Signed", vec![bytes]);
+    })
+}
+
+#[test]
+fn return_sender() {
+    with_executor(&|mut executor| {
+        let mut harness = deploy_contract(&mut executor, "return_sender.vy", "Foo");
+
+        let sender = address_token("1234000000000000000000000000000000005678");
+
+        harness.caller = sender.clone().to_address().unwrap();
+
+        harness.test_function(
+            &mut executor,
+            "bar",
+            // FIXME: There's an issue parsing functions with 0 params. This is here to mitigate that failure.
```

Comment:
general comment on `FIXME` -- if they are going to remain in the codebase for some indefinite period of time, we should move them to issues before we merge in this PR

Reply:
Alright, I'll create some issues for these comment.